### DISCUSSION TITLE
Game Entities

### DISCUSSION THREAD

**Discussion by fredchristianson** [fredchristianson/javascript-game-engine#9] [Ideas]
We can discuss the different entities (items, objects, actors) we may encounter in games.
- cards
- dice
- tiles
- animals
- trees
They will have a look that may be static or dynamic.  It may change over time, or location, or the world environment (e.g. lights)
They also may be inanimate or animate.  Animated entities could be changed by user action or computer controlled.

They may be drawn with html/css.  Or on a canvas.  Or SVG.

They may have 2 dimensions or 3.  And either type may be in a 2-dimensional or 3-dimensional world.

Understanding what attributes we want to support is necessary to architect the right classes and operations in the core.

**Comment by Eddie7145** [comment]
Loving this so that each world be it 2 or 3 dimensional would have its own immersive experience.

So my question now is that will the user be given the choice to pick a world, or will it be more of like a transgressing experience where after each level is passed it would take the user to the a different world?

**Comment by chylinski82** [comment]
- static entities: rocks, buildings, rivers, playing table/ board;
- dynamic entities: vehicles, enemies, balls, bullets, etc. Question about dynamic entities: do we create multiple angle of the same actor/sprite and then create animations, with some tool like Gimp? Dynamic entities should be able to move, jump, fall (gravity in physics?), shoot, climb, etc.
